src/common/store-update-message-hub.ts

```diff
@@ -22,14 +22,14 @@ export class StoreUpdateMessageHub {
         this.registeredUpdateListeners[storeId] = listener;
     }
 
-    public readonly handleMessage = (message: StoreUpdateMessage<any>): void => {
+    public readonly handleMessage = (message: StoreUpdateMessage<any>): void | Promise<void> => {
         if (!this.isValidMessage(message)) {
             return;
         }
 
         const listener = this.registeredUpdateListeners[message.storeId];
         if (listener) {
-            listener(message);
+            return Promise.resolve(listener(message));
```

Comment:
I think this should probably return the `listener` result directly, rather than wrapping in a `Promise.resolve`; wrapping in a promise will mask cases where a fire-and-forget listener hasn't been updated yet